eng/scripts/distribute-packages-to-matrix.ps1

```diff
@@ -0,0 +1,263 @@
+<#
+.SYNOPSIS
+Distributes a set of packages to a platform matrix file by adding computing and adding a TargetingSTring property
+dynamically. If an unexpected situation is encountered, the script will make no changes to the input file.
+
+.DESCRIPTION
+Because of the way the platform matrix file is structured, we need to distribute the packages in a way that
+honors the Include packages. We have these included this way because want to ensure that these python versions
+run on that platform. This script is aware of these additional configurations and will set TargetingString for
+them as well.
+#>
+param(
+    [parameter(Mandatory=$true)]
+    [string]$PackageInfoFolder,
+    [parameter(Mandatory=$true)]
+    [string]$PlatformMatrix
+)
+
+$RepoRoot = (Resolve-Path (Join-Path $PSScriptRoot ".." ".."))
+Set-StrictMode -Version 4
+$BATCHSIZE = 8
```

Comment:
follow-up: How did we choose this number? It might work for now but I suspect we might want to flip this into choose the number of agents we want to consume and then divide the projects based on that.

Reply:
This is an arbitrarily chosen number based on runtime. 8 packages worth of tests is ~30 to 40 minutes of runtime.